### PR TITLE
Add functionality to pass flag to register holoport for scalability testing

### PR DESCRIPTION
**Pull request by alastairong**
I will test this on a holoport in parallel.

I didn't write any tests since this is for internal use only so doesn't need the same level of reliability

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-stable
```
Comment (Br1ght0ne) [inline rust-toolchain:1]: What change requires `nightly`?

File: src/lib.rs [hunk 2]
```diff
@@ -30,8 +30,61 @@ use holochain_zome_types::{
...
 type HappIds = Vec<String>;
 
-pub async fn activate_holo_hosted_happs(core_happ: Happ) -> Result<()> {
-    let list_of_happs = get_enabled_hosted_happs(core_happ).await?;
+use ed25519_dalek::{PublicKey, SecretKey};
+use hpos_config_core::*;
+
+pub async fn handle_test_network_registration(core_happ: &Happ) -> Result<()> {
+    let hpos_config_path = env::var("HPOS_CONFIG_PATH")?;
+    let is_test_network = env::var("IS_TEST_NETWORK").is_err();
```
Comment (Br1ght0ne) [inline src/lib.rs:38]: So `is_test_network` is `true` when `$IS_TEST_NETWORK` is not set? e.g. the intended usecase for `$IS_TEST_NETWORK` is only `IS_TEST_NETWORK=false`?

File: src/lib.rs [hunk 3]
```diff
@@ -30,8 +30,61 @@ use holochain_zome_types::{
...
+    let host_id = get_host_id(hpos_config_path)?;
+    let list_of_happs = get_enabled_hosted_happs(&core_happ).await?;
+    update_test_network(list_of_happs, host_id, is_test_network).await?;
+    Ok(())
+}
+
+pub fn get_host_id(config_path: String) -> Result<String> {
+
+    let contents = fs::read(&config_path)?;
+    let hpos_config_core::Config::V1 { seed, .. } = serde_json::from_slice(&contents)?;
```
Comment (Br1ght0ne) [inline src/lib.rs:48]: Does this work with an enum variant?

**Review by Br1ght0ne — COMMENTED** [review]
Nice PR @alastairong! I like the readability, though I'd prefer for your changes to be `rustfmt`-d :)

I ask a couple of questions in the review, but they're not requests for changes - I just want to know your code better.

File: rust-toolchain (hunk 1 above)
Comment (timotree3) [reply rust-toolchain:1]: Using hpos-config-core

Comment (alastairong) [reply rust-toolchain:1]: It was an `ed25519-dalek` dependency. I forget which one.

File: src/lib.rs (hunk 2 above)
Comment (alastairong) [reply src/lib.rs:38]: Oops! My mistake cargo culting example code :)

File: src/lib.rs (hunk 3 above)
Comment (Br1ght0ne) [reply src/lib.rs:50]: Apparently, this works when an enum has only one variant. Cool!

**Review by Br1ght0ne — APPROVED** [review]
Dog LGTM Keyboard

**Pull request merged**